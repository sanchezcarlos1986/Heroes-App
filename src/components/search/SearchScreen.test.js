import React from 'react';
import {SearchScreen} from '~components';
import '@testing-library/jest-dom';
import {mount} from 'enzyme';
import {MemoryRouter, Route} from 'react-router-dom';

describe('<SearchScreen />', () => {
  let component;

  it('should renders default values', () => {
    component = mount(
        <MemoryRouter initialEntries={['/search']}>
          <Route path="/search" component={SearchScreen} />
        </MemoryRouter>,
    );
    expect(component).not.toBeNull();
    expect(component).toMatchSnapshot();
    expect(component.find('.alert-info').text()).toBe('Search a hero');
    expect(component.find('input').prop('value')).toBe('');
  });

  it('should renders could not find hero', () => {
    const search = 'ロックマン';

    component = mount(
        <MemoryRouter initialEntries={[`/search?q=${search}`]}>
          <Route path="/search" component={SearchScreen} />
        </MemoryRouter>,
    );
    expect(component.find('input').prop('value')).toBe(search);
    expect(component.find('.alert-danger').text()).toBe(
        `We couldn't find ${search}`,
    );
    expect(component.find('HeroCard').exists()).toBe(false);
  });

  it('should renders batman result', () => {
    const search = 'batman';

    component = mount(
        <MemoryRouter initialEntries={[`/search?q=${search}`]}>
          <Route path="/search" component={SearchScreen} />
        </MemoryRouter>,
    );
    expect(component.find('input').prop('value')).toBe(search);
    expect(component.find('HeroCard').exists()).toBe(true);
  });

  it('should send form and calls history PUSH', () => {
    const history = {
      push: jest.fn(),
    };

    const search = 'batman';
    const newSearch = 'spiderman';
    const fakeEvent = {preventDefault: () => jest.fn()};

    component = mount(
        <MemoryRouter initialEntries={[`/search?q=${search}`]}>
          <Route
            path="/search"
            component={() => <SearchScreen history={history} />}
          />
        </MemoryRouter>,
    );
    expect(component.find('input').prop('value')).toBe(search);

    component
        .find('input')
        .simulate('change', {target: {name: 'searchText', value: newSearch}});

    expect(component.find('input').prop('value')).toBe(newSearch);

    component.find('form').simulate('submit', fakeEvent);
    expect(history.push).toHaveBeenCalledWith(`?q=${newSearch}`);
  });
});
