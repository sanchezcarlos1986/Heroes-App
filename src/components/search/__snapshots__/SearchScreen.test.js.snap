// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchScreen /> should renders default values 1`] = `
<div>
  <h1>
    SearchScreen
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-5"
    >
      <h4>
        Search Form
      </h4>
      <hr />
      <form
        onSubmit={[Function]}
      >
        <input
          autoComplete="off"
          className="form-control"
          name="searchText"
          onChange={[Function]}
          placeholder="Find your hero!"
          type="text"
          value=""
        />
        <button
          className="btn m-t1 btn-block btn-outline-primary"
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
    <div
      className="col-7"
    >
      <h4>
        Your heroes
      </h4>
      <div
        className="alert alert-info"
      >
        Search a hero
      </div>
    </div>
  </div>
</div>
`;
